packages/react-core/package.json

```diff
@@ -21,7 +21,7 @@
   },
   "devDependencies": {
     "@types/history": "^4.7.9",
-    "@types/react": "^17.0.31",
+    "@types/react": "^18.0.0",
     "@types/react-dom": "^17.0.10",
```

Comment:
Same here - should we upgrade react-dom types since we're updating the peer-dep range?